sqlglot/parser.py

```diff
@@ -946,14 +949,29 @@ def _parse_create(self) -> t.Optional[exp.Expression]:
             this = self._parse_table(schema=True)
             properties = self._parse_properties()
             if self._match(TokenType.ALIAS):
-                expression = self._parse_select(nested=True)
+                expression = self._parse_select(nested=True, parse_subquery_alias=False)
 
             if create_token.token_type == TokenType.TABLE:
                 if self._match_text_seq("WITH", "DATA"):
                     data = True
                 elif self._match_text_seq("WITH", "NO", "DATA"):
                     data = False
 
+                if self._match_text_seq("AND", "STATISTICS"):
+                    statistics = True
+                elif self._match_text_seq("AND", "NO", "STATISTICS"):
+                    statistics = False
+
+                no_primary_index = self._match_text_seq("NO", "PRIMARY", "INDEX")
+
+                indexes = []
+                while (
```

Comment:
maybe this should be

while (self._curr and self._curr.text.upper() in ("UNIQUE", "PRIMARY", "INDEX) 

or make UNIQUE, PRIMARY, INDEX in a constant,

then you won't need that code in the create_table_index section

that way you don't advance

Reply:
do you need this condition at all? could you do

while True:
  self._parse_create_table_index()

   if not index:
      break

Reply:
Yes, that's a much better solution!